src/utils/ProteomicsLFQ.cpp

```diff
@@ -0,0 +1,1598 @@
+#include <OpenMS/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.h>
+#include <OpenMS/FILTERING/DATAREDUCTION/FeatureFindingMetabo.h>
+#include <OpenMS/ANALYSIS/MAPMATCHING/FeatureGroupingAlgorithmQT.h>
+//#include <OpenMS/ANALYSIS/MAPMATCHING/FeatureGroupingAlgorithmKD.h>
+
+#include <OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmIdentification.h>
+#include <OpenMS/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.h>
+#include <OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentTransformer.h>
+#include <OpenMS/ANALYSIS/ID/IDConflictResolverAlgorithm.h>
+#include <OpenMS/ANALYSIS/ID/BasicProteinInferenceAlgorithm.h>
+#include <OpenMS/ANALYSIS/ID/BayesianProteinInferenceAlgorithm.h>
+#include <OpenMS/ANALYSIS/ID/FalseDiscoveryRate.h>
+#include <OpenMS/ANALYSIS/ID/IDBoostGraph.h>
+#include <OpenMS/ANALYSIS/ID/PeptideProteinResolution.h>
+#include <OpenMS/ANALYSIS/MAPMATCHING/ConsensusMapNormalizerAlgorithmMedian.h>
+#include <OpenMS/FILTERING/DATAREDUCTION/ElutionPeakDetection.h>
+
+
+#include <OpenMS/FILTERING/TRANSFORMERS/ThresholdMower.h>
+
+#include <OpenMS/FORMAT/MzTabFile.h>
+#include <OpenMS/FORMAT/MzTab.h>
+#include <OpenMS/FORMAT/IdXMLFile.h>
+#include <OpenMS/FORMAT/ExperimentalDesignFile.h>
+#include <OpenMS/METADATA/SpectrumMetaDataLookup.h>
+
+#include <OpenMS/KERNEL/ConversionHelper.h>
+
+#include <OpenMS/FORMAT/DATAACCESS/MSDataWritingConsumer.h>
+#include <OpenMS/KERNEL/MassTrace.h>
+#include <OpenMS/FILTERING/DATAREDUCTION/MassTraceDetection.h>
+
+#include <OpenMS/FILTERING/ID/IDFilter.h>
+
+#include <OpenMS/ANALYSIS/ID/PeptideIndexing.h>
+#include <OpenMS/ANALYSIS/ID/IDMergerAlgorithm.h>
+
+using namespace OpenMS;
+using namespace std;
+
+//-------------------------------------------------------------
+//Doxygen docu
+//-------------------------------------------------------------
+
+/**
+  @page UTILS_ProteomicsLFQ
+
+  // experiments TODO:
+  // - change percentage of missingness in ID transfer
+  // - disable elution peak fit
+
+  // TODO:
+     - check experimental design at startup
+     - run R script directly from tool?
+
+  TODO: add test for
+
+  ./bin/ProteomicsLFQ \
+-in ../share/OpenMS/examples/FRACTIONS/BSA1_F1.mzML ../share/OpenMS/examples/FRACTIONS/BSA1_F2.mzML ../share/OpenMS/examples/FRACTIONS/BSA2_F1.mzML ../share/OpenMS/examples/FRACTIONS/BSA2_F2.mzML ../share/OpenMS/examples/FRACTIONS/BSA3_F1.mzML ../share/OpenMS/examples/FRACTIONS/BSA3_F2.mzML \
+-ids ../share/OpenMS/examples/FRACTIONS/BSA1_F1.idXML ../share/OpenMS/examples/FRACTIONS/BSA1_F2.idXML ../share/OpenMS/examples/FRACTIONS/BSA2_F1.idXML ../share/OpenMS/examples/FRACTIONS/BSA2_F2.idXML ../share/OpenMS/examples/FRACTIONS/BSA3_F1.idXML ../share/OpenMS/examples/FRACTIONS/BSA3_F2.idXML \
+-design ../share/OpenMS/examples/FRACTIONS/BSA_design.tsv \
+-Alignment:max_rt_shift 0 \
+-fasta ../share/OpenMS/examples/TOPPAS/data/BSA_Identification/18Protein_SoCe_Tr_detergents_trace_target_decoy.fasta \
+-out BSA.mzTab -threads 4 -debug 667 
```

Comment:
can you clean this up? all this will show up in the doxygen docu otherwise?

Reply:
PS: also tests are currently missing?

Reply:
yes kind of missing. There are scripts in the test folder to run it on BSA. But it is not in the test suite yet. We will check how to best do this.
I'll clean up the comments.